src/lightning/fabric/strategies/strategy.py

```diff
@@ -305,6 +306,39 @@ def teardown(self) -> None:
         self.accelerator.teardown()
         self.checkpoint_io.teardown()
 
+    def _unscale_params(self, optimizer):
+        scaler = getattr(self.precision, "scaler", None)
+        if scaler is not None:
+            if _optimizer_handles_unscaling(optimizer):
+                raise NotImplementedError("Gradient clipping is not implemented for optimizers handling the unscaling.")
+            scaler.unscale_(optimizer)
+
+    def clip_gradients_norm(
```

Comment:
IMO, the base implementation should be in the precision plugin base, and the strategy here just calls `self.precision.clip_...`

A strategy implementing the gradient clipping can decide whether to override or call into its own precision plugin.

Reply:
I disagree here. IMO this should live in the strategy itself as it's not related to precision handling. Yes, it may need to unscale gradients, but that's all there is. And that's exactly why the precision plugin lives in the strategy now and they aren't on the same level within the accelerator anymore.

This is especially true as some strategies (e.g. FSDP) require collective communications here and I don't like the implementations of this to be scattered between precision plugins and strategies.

Reply:
after discussing offline with @awaelchli we agreed to move `unscale_params` as `unscale_gradients` to the precision plugin which can then be called here.